smack-experimental/src/main/java/org/jivesoftware/smackx/sid/element/OriginIdElement.java

```diff
@@ -101,4 +101,25 @@ public CharSequence toXML(org.jivesoftware.smack.packet.XmlEnvironment enclosing
                 .attribute(ATTR_ID, getId())
                 .closeEmptyElement();
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null) {
+            return false;
+        }
+        if (!(other instanceof OriginIdElement)) {
+            return false;
+        }
+
+        OriginIdElement otherId = (OriginIdElement) other;
+        return getId().equals(otherId.getId());
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Similarly, there is `HashCode` class inside smack.util package.

Reply:
I wasn't aware of that class either :P